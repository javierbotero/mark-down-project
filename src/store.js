import { configureStore, createSlice } from '@reduxjs/toolkit';
import { initialStateText } from './utils/utils';

const textReducer = createSlice({
  name: 'text',
  initialState: initialStateText,
  reducers: {
    addText(state, action) { return action.payload }
  }
});
const store = configureStore({
  reducer: {
    text: textReducer.reducer
  }
});

export default store;
export const { addText } = textReducer.actions;
