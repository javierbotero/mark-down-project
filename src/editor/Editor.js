import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Preview from '../preview/Preview';
import { addText } from '../store';

const Editor = () => {
  const dispatch = useDispatch();
  const text = useSelector(state => state.text)
  const onChange = (event) => dispatch(addText(event.target.value));
  return (
    <>
      <textarea
        id="editor"
        placeholder="placeholder"
        onChange={onChange}
        value={text}
        rows="10"
        className="col-12 forty-percent-height border border-5"
      />
      <Preview />
    </>
  )
}

export default Editor;
