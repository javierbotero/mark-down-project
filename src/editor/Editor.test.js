import renderer from 'react-test-renderer';
import Editor from './Editor';
import { Provider } from 'react-redux';
import store from '../store';

it('checks print of component', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Editor />
    </Provider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})