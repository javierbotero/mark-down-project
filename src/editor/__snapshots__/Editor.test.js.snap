// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check print of component 1`] = `
Array [
  <textarea
    id="editor"
    onChange={[Function]}
    placeholder="placeholder"
    rows="10"
    value="
# Welcome to my React Markdown Previewer!

## This is a sub-heading...
### And here's some other cool stuff:

Heres some code, \`<div></div>\`, between 2 backticks.

\`\`\`
// this is multi-line code:

function anotherExample(firstLine, lastLine) {
  if (firstLine == '\`\`\`' && lastLine == '\`\`\`') {
    return multiLineCode;
  }
}
\`\`\`

You can also make text **bold**... whoa!
Or _italic_.
Or... wait for it... **_both!_**
And feel free to go crazy ~~crossing stuff out~~.

There's also [links](https://www.freecodecamp.org), and
> Block Quotes!

And if you want to get really crazy, even tables:

Wild Header | Crazy Header | Another Header?
------------ | ------------- | -------------
Your content can | be here, and it | can be here....
And here. | Okay. | I think we get it.

- And of course there are lists.
  - Some are bulleted.
     - With different indentation levels.
        - That look like this.


1. And there are numbered lists too.
1. Use just 1s if you want!
1. And last but not least, let's not forget embedded images:

![freeCodeCamp Logo](https://cdn.freecodecamp.org/testable-projects-fcc/images/fcc_secondary.svg)
"
  />,
  <div
    className="py-5 not-overflow"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h1 id=\\"welcome-to-my-react-markdown-previewer\\">Welcome to my React Markdown Previewer!</h1>
<h2 id=\\"this-is-a-sub-heading\\">This is a sub-heading...</h2>
<h3 id=\\"and-heres-some-other-cool-stuff\\">And here's some other cool stuff:</h3>
<p>Heres some code, <code>&lt;div&gt;&lt;/div&gt;</code>, between 2 backticks.</p>
<pre><code>// this is multi-line code:

function anotherExample(firstLine, lastLine) {
  if (firstLine == '\`\`\`' &amp;&amp; lastLine == '\`\`\`') {
    return multiLineCode;
  }
}
</code></pre>
<p>You can also make text <strong>bold</strong>... whoa!<br>Or <em>italic</em>.<br>Or... wait for it... <strong><em>both!</em></strong><br>And feel free to go crazy <del>crossing stuff out</del>.</p>
<p>There's also <a href=\\"https://www.freecodecamp.org\\">links</a>, and</p>
<blockquote>
<p>Block Quotes!</p>
</blockquote>
<p>And if you want to get really crazy, even tables:</p>
<table>
<thead>
<tr>
<th>Wild Header</th>
<th>Crazy Header</th>
<th>Another Header?</th>
</tr>
</thead>
<tbody><tr>
<td>Your content can</td>
<td>be here, and it</td>
<td>can be here....</td>
</tr>
<tr>
<td>And here.</td>
<td>Okay.</td>
<td>I think we get it.</td>
</tr>
</tbody></table>
<ul>
<li>And of course there are lists.<ul>
<li>Some are bulleted.<ul>
<li>With different indentation levels.<ul>
<li>That look like this.</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<ol>
<li>And there are numbered lists too.</li>
<li>Use just 1s if you want!</li>
<li>And last but not least, let's not forget embedded images:</li>
</ol>
<p><img alt=\\"freeCodeCamp Logo\\" src=\\"https://cdn.freecodecamp.org/testable-projects-fcc/images/fcc_secondary.svg\\"></p>
",
      }
    }
    id="preview"
  />,
]
`;

exports[`checks print of component 1`] = `
Array [
  <textarea
    className="col-12"
    id="editor"
    onChange={[Function]}
    placeholder="placeholder"
    rows="10"
    value="
# Welcome to my React Markdown Previewer!

## This is a sub-heading...
### And here's some other cool stuff:

Heres some code, \`<div></div>\`, between 2 backticks.

\`\`\`
// this is multi-line code:

function anotherExample(firstLine, lastLine) {
  if (firstLine == '\`\`\`' && lastLine == '\`\`\`') {
    return multiLineCode;
  }
}
\`\`\`

You can also make text **bold**... whoa!
Or _italic_.
Or... wait for it... **_both!_**
And feel free to go crazy ~~crossing stuff out~~.

There's also [links](https://www.freecodecamp.org), and
> Block Quotes!

And if you want to get really crazy, even tables:

Wild Header | Crazy Header | Another Header?
------------ | ------------- | -------------
Your content can | be here, and it | can be here....
And here. | Okay. | I think we get it.

- And of course there are lists.
  - Some are bulleted.
     - With different indentation levels.
        - That look like this.


1. And there are numbered lists too.
1. Use just 1s if you want!
1. And last but not least, let's not forget embedded images:

![freeCodeCamp Logo](https://cdn.freecodecamp.org/testable-projects-fcc/images/fcc_secondary.svg)
"
  />,
  <div
    className="col-12 py-5 not-overflow"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h1 id=\\"welcome-to-my-react-markdown-previewer\\">Welcome to my React Markdown Previewer!</h1>
<h2 id=\\"this-is-a-sub-heading\\">This is a sub-heading...</h2>
<h3 id=\\"and-heres-some-other-cool-stuff\\">And here's some other cool stuff:</h3>
<p>Heres some code, <code>&lt;div&gt;&lt;/div&gt;</code>, between 2 backticks.</p>
<pre><code>// this is multi-line code:

function anotherExample(firstLine, lastLine) {
  if (firstLine == '\`\`\`' &amp;&amp; lastLine == '\`\`\`') {
    return multiLineCode;
  }
}
</code></pre>
<p>You can also make text <strong>bold</strong>... whoa!<br>Or <em>italic</em>.<br>Or... wait for it... <strong><em>both!</em></strong><br>And feel free to go crazy <del>crossing stuff out</del>.</p>
<p>There's also <a href=\\"https://www.freecodecamp.org\\">links</a>, and</p>
<blockquote>
<p>Block Quotes!</p>
</blockquote>
<p>And if you want to get really crazy, even tables:</p>
<table>
<thead>
<tr>
<th>Wild Header</th>
<th>Crazy Header</th>
<th>Another Header?</th>
</tr>
</thead>
<tbody><tr>
<td>Your content can</td>
<td>be here, and it</td>
<td>can be here....</td>
</tr>
<tr>
<td>And here.</td>
<td>Okay.</td>
<td>I think we get it.</td>
</tr>
</tbody></table>
<ul>
<li>And of course there are lists.<ul>
<li>Some are bulleted.<ul>
<li>With different indentation levels.<ul>
<li>That look like this.</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<ol>
<li>And there are numbered lists too.</li>
<li>Use just 1s if you want!</li>
<li>And last but not least, let's not forget embedded images:</li>
</ol>
<p><img alt=\\"freeCodeCamp Logo\\" src=\\"https://cdn.freecodecamp.org/testable-projects-fcc/images/fcc_secondary.svg\\"></p>
",
      }
    }
    id="preview"
  />,
]
`;
